Disable use of HAVE(36BIT_ADDRESS).

MenloDorian opened the pull request:
#### 06d5550e6d760d556d785bbdb21f3c2a49b8f68b
<pre>
Disable use of HAVE(36BIT_ADDRESS).
<a href="https://bugs.webkit.org/show_bug.cgi?id=304052">https://bugs.webkit.org/show_bug.cgi?id=304052</a>
<a href="https://rdar.apple.com/166369379">rdar://166369379</a>

Reviewed by Keith Miller.

We cannot guarantee that the kernel will always map pages with 36-bit addresses.  As such, we
cannot use HAVE(36BIT_ADDRESS).  For now, we will simply disable the use of HAVE(36BIT_ADDRESS)
instead of removing the code completely.  We will completely remove the HAVE(36BIT_ADDRESS)
code after we have some baking time.

Also clean up or fix some uses of EFFECTIVE_ADDRESS_WIDTH that fall out of disabling
HAVE(36BIT_ADDRESS).  This includes:

1. Removed tagArrayPtr, untagArrayPtr, untagArrayPtrLength64, and removeArrayPtrTag functions
   from MacroAssemblerARM64E.h.  These functions are now unused.

   With that, we can remove the maxNumberOfAllowedPACBits constant which relies on
   OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH), which ultimately relies on HAVE(36BIT_ADDRESS).
   This constant was previously being used by removeArrayPtrTag.  The number of PAC bits is
   not ABI.  Hence, it is generally not ABI safe / stable to rely on this.  Hence, I opted
   to just remove this code since they are unused anyway.

2. Removed some unused constants in CagePtr.h that rely on OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH).

3. Fix hasCapacityToUseLargeGigacage to not rely on OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH).
   OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH) may not be a good signal on whether we can afford
   to use a large Gigacage or not.  Instead, just set this value based on whether we&apos;re
   targetting an embedded platform, or not.

   Also set hasCapacityToUseLargeGigacage to false for CPU(ADDRESS32) just for consistency.
   Note: CPU(ADDRESS32) platforms do not actually enable Gigacage.

No new tests.  This is just a feature disablement.  The affected code will already be
covered by existing tests.

* Source/JavaScriptCore/assembler/MacroAssemblerARM64E.h:
(JSC::MacroAssemblerARM64E::tagArrayPtr): Deleted.
(JSC::MacroAssemblerARM64E::untagArrayPtr): Deleted.
(JSC::MacroAssemblerARM64E::untagArrayPtrLength64): Deleted.
(JSC::MacroAssemblerARM64E::removeArrayPtrTag): Deleted.
* Source/WTF/wtf/CagedPtr.h:
* Source/WTF/wtf/Gigacage.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/bmalloc/bmalloc/BPlatform.h:
* Source/bmalloc/bmalloc/Gigacage.h:
* Source/bmalloc/libpas/src/libpas/pas_platform.h:

Canonical link: <a href="https://commits.webkit.org/304363@main">https://commits.webkit.org/304363@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ef79130fedede3e94e56e53f6bc22c1e594b7c0e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135338 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7724 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46609 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142842 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/87067 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/175042f1-6e6d-4538-b9d7-6066554a72db) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8356 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7565 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103428 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/70802 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f8875ef1-7f56-4091-bbbe-7e86038d0efe) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138284 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5978 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121298 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84293 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7054da03-0c74-44ec-8fc3-7c9ff1c435c2) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5762 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3363 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3428 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/127333 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114967 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39481 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145537 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/133822 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7402 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40050 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111804 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7444 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6200 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112173 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5611 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117598 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/61323 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20873 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7457 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35724 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/166667 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7207 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71003 "Built successfully") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/43543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7427 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7307 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/JavaScriptCore/assembler/MacroAssemblerARM64E.h:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2018-2023 Apple Inc. All rights reserved.
+ * Copyright (C) 2018-2025 Apple Inc. All rights reserved.
```
I think we’re supposed to keep all the years. Ditto for the other files.

kmiller68 replied:
Oh sorry my GH UI showed this as only having 2023 before not 2018-2023. Ignore me.

kmiller68 reviewed (approved):
R=me with copyright fixes

Commit:
Disable use of HAVE(36BIT_ADDRESS).
https://bugs.webkit.org/show_bug.cgi?id=304052
rdar://166369379

Reviewed by Keith Miller.

We cannot guarantee that the kernel will always map pages with 36-bit addresses.  As such, we
cannot use HAVE(36BIT_ADDRESS).  For now, we will simply disable the use of HAVE(36BIT_ADDRESS)
instead of removing the code completely.  We will completely remove the HAVE(36BIT_ADDRESS)
code after we have some baking time.

Also clean up or fix some uses of EFFECTIVE_ADDRESS_WIDTH that fall out of disabling
HAVE(36BIT_ADDRESS).  This includes:

1. Removed tagArrayPtr, untagArrayPtr, untagArrayPtrLength64, and removeArrayPtrTag functions
   from MacroAssemblerARM64E.h.  These functions are now unused.

   With that, we can remove the maxNumberOfAllowedPACBits constant which relies on
   OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH), which ultimately relies on HAVE(36BIT_ADDRESS).
   This constant was previously being used by removeArrayPtrTag.  The number of PAC bits is
   not ABI.  Hence, it is generally not ABI safe / stable to rely on this.  Hence, I opted
   to just remove this code since they are unused anyway.

2. Removed some unused constants in CagePtr.h that rely on OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH).

3. Fix hasCapacityToUseLargeGigacage to not rely on OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH).
   OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH) may not be a good signal on whether we can afford
   to use a large Gigacage or not.  Instead, just set this value based on whether we're
   targetting an embedded platform, or not.

   Also set hasCapacityToUseLargeGigacage to false for CPU(ADDRESS32) just for consistency.
   Note: CPU(ADDRESS32) platforms do not actually enable Gigacage.

No new tests.  This is just a feature disablement.  The affected code will already be
covered by existing tests.

* Source/JavaScriptCore/assembler/MacroAssemblerARM64E.h:
(JSC::MacroAssemblerARM64E::tagArrayPtr): Deleted.
(JSC::MacroAssemblerARM64E::untagArrayPtr): Deleted.
(JSC::MacroAssemblerARM64E::untagArrayPtrLength64): Deleted.
(JSC::MacroAssemblerARM64E::removeArrayPtrTag): Deleted.
* Source/WTF/wtf/CagedPtr.h:
* Source/WTF/wtf/Gigacage.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/bmalloc/bmalloc/BPlatform.h:
* Source/bmalloc/bmalloc/Gigacage.h:
* Source/bmalloc/libpas/src/libpas/pas_platform.h:

Canonical link: https://commits.webkit.org/304363@main
